x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -980,12 +981,16 @@ ApiKeyCredentials getCredentialsFromHeader(ThreadContext threadContext) {
         return getCredentialsFromHeader(threadContext.getHeader("Authorization"));
     }
 
-    ApiKeyCredentials getCredentialsFromHeader(final String header) {
-        assert isEnabled() : "API keys must be enabled";
```

Comment:
This assertion is dropped. Is it necessary?

Reply:
It allows us to use this method statically in RemoteAccessHeaders, without passing an ApiKeyService instance. I think it's fair enough to decouple parsing functionality from checking if the service is enabled (we still make the enabled check in `RemoteAccessAuthenticationService`, to ensure a friendly error message).